qemu/tests/guest_iommu_group.py

```diff
@@ -0,0 +1,35 @@
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    this case will:
+    1) Boot guest with virtio devices and iommu is on.
+    2) Check whether the iommu group is seperated correctly.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def verify_iommu_group():
+        """ Verify whether the iommu group is seperated correctly. """
+        error_context.context(
+            'Verify whether the iommu group is seperated correctly.', test.log.info)
+        device_id = session.cmd("lspci | grep 'PCIe\\|Virtio' | awk '{print $1}'").strip().split()
+        group_id = []
+        for id in device_id:
+            group_id.append(session.cmd("dmesg | grep \"iommu group\" | grep '%s' | awk -F \" \" '{print $NF}'" % ("0000:" + id)).strip())
+        test.log.info("group_id: %s", group_id)
```

Comment:
If you want to print the debug info, better to print the device id/name as well. `Group ID of xxxx: %s`